src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -323,35 +328,106 @@ FancyZones::VirtualDesktopInitialize() noexcept
     PostMessage(m_window, WM_PRIV_VD_INIT, 0, 0);
 }
 
+bool FancyZones::ShouldProcessNewWindow(HWND window) noexcept
+{
+    // Avoid processing splash screens, already stamped (zoned) windows, or those windows
+    // that belong to excluded applications list.
+    if (IsSplashScreen(window) ||
+        (reinterpret_cast<size_t>(::GetProp(window, MULTI_ZONE_STAMP)) != 0) ||
+        !IsInterestingWindow(window, m_settings->GetSettings()->excludedAppsArray))
+    {
+        return false;
+    }
+    return true;
+}
+
+std::pair<winrt::com_ptr<IZoneWindow>, std::vector<int>> FancyZones::GetZoneIndexSetFromWorkAreaHistory(
+    HWND window, winrt::com_ptr<IZoneWindow> workArea)

```

Comment:
maybe to pass an entire map as arg, and rename method to `SearchAppHistory` or something similar instead of returning a passed argument ?

Reply:
or just return vector and append the workarea in caller function

Reply:
I'll just return vector